### PR TITLE
HPCC-20719 Fix 2 problems in ESP Transaction Logging

### PR DESCRIPTION
**Pull request by wangkx**
1. the filterLogContent() should not overwrite original logging
request because other logging agent may use it;
2. wsloggingserviceagent.xsl should only create the setting for
one agent instance because multiple agent instances have been
handled by loggingmanager related xsl.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
This PR should be reviewed/merged with https://github.com/hpcc-systems/LN/pull/893.

File: esp/logging/logginglib/loggingagentbase.cpp [hunk 1]
```diff
@@ -294,8 +294,8 @@ void CLogContentFilter::filterLogContent(IEspUpdateLogRequestWrap* req)
     StringBuffer updateLogRequestXML;
     toXML(updateLogRequestTree, updateLogRequestXML);
     ESPLOG(LogMax, "filtered content and option: <%s>", updateLogRequestXML.str());
-    req->clearOriginalContent();
-    req->setUpdateLogRequest(updateLogRequestXML.str());
+
+    return new CUpdateLogRequestWrap(req->getGUID(), req->getOption(), updateLogRequestXML.str());
```
Comment (asselitx) [inline esp/logging/logginglib/loggingagentbase.cpp]: It doesn't appear that CLogThread::queueLog takes ownership of the new CUpdatelogRequestWrap when it calls filterLogContent. Is this a possible leak, or am I missing something?

File: initfiles/componentfiles/configxml/@temp/wslogserviceespagent.xsl [hunk 2]
```diff
@@ -23,100 +23,97 @@ xmlns:set="http://exslt.org/sets">
...
+        <xsl:if test="not($agentNode)">
+            <xsl:message terminate="yes">An WsLogService ESP Logging Agent <xsl:value-of select="$agentName"/> is undefined!</xsl:message>
+        </xsl:if>
+        <xsl:variable name="loggingServer" select="$agentNode/LoggingServer"/>
+        <xsl:if test="not($loggingServer)">
+            <xsl:message terminate="yes">ESP logging server is undefined for <xsl:value-of select="$agentName"/> </xsl:message>
+        </xsl:if>
+        <xsl:variable name="loggingServerUrl" select="$loggingServer/@Url"/>
+        <xsl:if test="string($loggingServerUrl) = ''">
+            <xsl:message terminate="yes">Logging Server URL is undefined!</xsl:message>
```
Comment (asselitx) [inline initfiles/componentfiles/configxml/@temp/wslogserviceespagent.xsl]: Perhaps also mention the $agentName here too for easy reference which agent has the error.

Comment (wangkx) [reply initfiles/componentfiles/configxml/@temp/wslogserviceespagent.xsl]: Good point. I will change it.

**Comment by richardkchapman** [comment]
Should this target 7.0.2?

**Comment by wangkx** [comment]
I will change it to 7.0.2. Still waiting for https://github.com/hpcc-systems/LN/pull/893.

**Comment by wangkx** [comment]
@asselitx please review my change.

**Comment by asselitx** [comment]
Reviewed, ready for merge.

**Comment by wangkx** [comment]
A new PR will be created for 7.0.2.

**Pull request closed without merging**